### PR TITLE
6354 os info win

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|#6354|

## **Description**

This issue aims to make use of the generic synchronization libraries, to use the synchronization solution algorithm that both shared modules provide.

These changes have to be made in the various implementations according to the operating system targeted for this feature.

## **DoD**
- [X] Implementation
- [X] **Run testtool on Win XP**
image
- [X] **Run testtool on Win 7**
image
image
- [X] **Run testtool on Win 10**
image
- **SysInfo UTs:**
> **Linux:**
image
> **Windows:**
image

### CODE REVIEW

File: src/wazuh_modules/syscollector/tests/sysInfo/sysOsInfo_test.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Wazuh SysInfoParsers
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * October 19, 2020.
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/tests/sysInfo/sysOsInfo_test.cpp:4]: Wrong header title and date

**Review by tinachoB — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/syscollector/sysInfoWin.cpp [hunk 2]
```diff
@@ -603,6 +604,18 @@ nlohmann::json SysInfo::getPackages() const
 
 nlohmann::json SysInfo::getOsInfo() const
 {
-    // Currently not supported for this OS
-    return {};
+    nlohmann::json ret;
+    const std::shared_ptr<ISysOsInfoProvider> spOsInfoProvider
+    {
+        new SysOsInfoProviderWindows
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/sysInfoWin.cpp:610]: use make_shared

File: src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        Utils::Registry currentVersion{HKEY_LOCAL_MACHINE, R"(SOFTWARE\Microsoft\Windows NT\CurrentVersion)"};
+        if (IsWindows8OrGreater())
+        {
+            const auto versionNumber{currentVersion.dword(isMinor ? "CurrentMinorVersionNumber" : "CurrentMajorVersionNumber")};
+            version = std::to_string(versionNumber);
+        }
+        else
+        {
+            const auto fullVersion{currentVersion.string("CurrentVersion")};
+            const auto majorAndMinor{Utils::split(fullVersion, '.')};
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp:34]: check if size is 2, to avoid exception for missing information.

Comment (Dwordcito) [reply src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp:34]: add enum OS_VERSION { MAJOR, MINOR, SIZE };

File: src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    }
+    return name.empty() ? "Microsoft Windows" : name;
+}
+
+static std::string getMachine()
+{
+    std::string machine{"unknown"};
+    Utils::Registry environment{HKEY_LOCAL_MACHINE, R"(System\CurrentControlSet\Control\Session Manager\Environment)"};
+    const auto arch{environment.string("PROCESSOR_ARCHITECTURE")};
+    if (arch == "AMD64" || arch == "IA64" || arch == "ARM64")
```
Comment (Dwordcito) [inline src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp:198]: add static map for this condition

Comment (Dwordcito) [reply src/wazuh_modules/syscollector/sysOsInfo/sysOsInfoWin.cpp:198]: in sharedDefs.h is a good place.

**Review by Dwordcito — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**